packages/terra-form-select/src/native-select/NativeSelect.jsx

```diff
@@ -0,0 +1,228 @@
+
+const cx = classNamesBind.bind(styles);
+
+const propTypes = {
+  /**
+   * The string containing ids for elements to describe the select.
+   */
+  ariaDescribedBy: PropTypes.string,
+  /**
+   * The aria label string value for the select.
+   */
+  ariaLabel: PropTypes.string.isRequired,
+  /**
+   * Additional attributes to spread onto the select.
+   */
+  // eslint-disable-next-line react/forbid-prop-types
+  attrs: PropTypes.object,
+  /**
+   * The default value of the select. Can be a string, or number.
+   */
+  defaultValue: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  /**
+   * Whether the input is disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * The Select identifier. Links the htmlFor of the field to the select identifier.
+   */
+  id: PropTypes.string.isRequired,
+  /**
+   * @private
+   * The intl object to be injected for translations.
+   */
+  intl: intlShape.isRequired,
+  /**
+   * Whether the input is invalid.
+   */
+  isInvalid: PropTypes.bool,
+  /**
+   * Whether the field displays as Incomplete. Use when no value has been provided. _(usage note: `required` must also be set)_.
+   */
+  isIncomplete: PropTypes.bool,
+  /**
+   * Callback function triggered when the select value changes. function(value)
+   */
+  onChange: PropTypes.func,
+  /**
+   * The array of select options and opt groups.
+   */
+  options: PropTypes.arrayOf(PropTypes.oneOfType([optionPropType, optGroupPropType])),
+  /**
+   * Placeholder data. The presence of this data object is used as the indicator for whether or not a placeholder is used.
+   */
+  placeholder: PropTypes.shape({
+    /**
+     * Whether the placeholder is included as an option to clear selection.
+     */
+    allowClear: PropTypes.bool,
+    /**
+     * Optional override of the default display.
+     */
+    display: PropTypes.string,
+    /**
+     * Optional override of the default value.
+     */
+    value: PropTypes.string,
+  }),
+  /**
+   * Whether the field is required.
+   */
+  required: PropTypes.bool,
+  /**
+   * The value of the select. Can be a string or number.
+   */
+  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+};
+
+const defaultProps = {
+  disabled: false,
+  isIncomplete: false,
+  isInvalid: false,
+  options: [],
+  required: false,
+};
+
+const createPlaceholder = (placeholder, intl) => {
+  if (!placeholder) {
+    return undefined;
+  }
+  const display = placeholder.display ? placeholder.display : intl.formatMessage({ id: 'Terra.form.select.defaultDisplay' });
+  const value = placeholder.value ? placeholder.value : defaultPlaceholderValue;
+  const attrs = placeholder.allowClear ? {} : { disabled: true, hidden: true };
+  return <option value={value} {...attrs}>{display}</option>;
+};
+
+const createOptions = options => options.map(current => {
+  if (current.childOptions) {
+    return (
+      <optgroup key={`${current.display} ${current.childOptions.length}`} label={current.display}>
```

Comment:
Are the various displays expected to be unique? It'd be really confusing if they weren't unique, but I'm not sure if that's explicitly called out anywhere.

Reply:
I believe the native select allows it, so I could adjust it do be a display + index. Thoughts?

Reply:
Yeah only index should be good here. even the option also can have the same instead of value below.

Reply:
Updated with index of key occurrence method.